File: week1/mario.c
#include <cs50.h>
#include <stdio.h>

void print_space(int height);
void print_brick(int height);
int main(void)
{
    int input;
    do
    {
        input = get_int("How high is the pyramid? (Between 1 and 8) ");
    }
    while (input < 1 || input > 8);

    int bricks = 1;
    do
    {
        print_space(input);
        print_brick(bricks);
        printf("  ");
        print_brick(bricks);
        input--;
        bricks++;
        printf("\n");
    }
    while (input > 0);
}

void print_space(int height)
{
    for (int i = 1; i < height; i++)
    {
        printf(" ");
    }
}

void print_brick(int height)
{
    for (int j = 0; j < height; j++)
    {
        printf("#");
    }
}

File: week6/mario.py
def main():

    height = validity()
    blank = height - 1
    bricks = 1

    for _ in range(height):
        print_line(blank, bricks)
        blank -= 1
        bricks += 1


def validity():
    while True:
        try:
            height = int(input("Enter pyramid height (1 - 8): "))
            if 1 <= height <= 8:
                return height
        except ValueError:
            print("Enter an integer between 1 and 8.")


def print_line(blank, bricks):
    for _ in range(blank):
        print(" ", end="")

    for _ in range(bricks):
        print("#", end="")

    print("  ", end="")

    for _ in range(bricks):
        print("#", end="")

    print("")


main()
